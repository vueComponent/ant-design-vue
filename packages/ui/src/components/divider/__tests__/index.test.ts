import { describe, expect, it, vi } from 'vitest'
import { Divider } from '@ant-design-vue/ui'
import { mount } from '@vue/test-utils'

describe('Divider', () => {
  it('should render correctly', () => {
    const wrapper = mount(Divider, {
      slots: {
        default: `test`
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  });

  it('not show children when vertical', () => {
    const wrapper = mount(Divider, {
      props: {
        orientation: 'right',
        orientationMargin: '10px',
      },
      slots: {
        default: `test`,
      },
    })

    expect(wrapper.find('.ant-divider-inner-text').exists()).toBe(true)
    const innerTextElement = wrapper.find('.ant-divider-inner-text')
    expect(innerTextElement.attributes('style')).toContain('margin-inline-end: 10px')
  })

  it('support bool dashed', () => { 
    const wrapper = mount(Divider, {
      props: {
        dashed: true,
      },
      slots: {
        default: `test`,
      },
    })
    expect(wrapper.find('.ant-divider-dashed').exists()).toBe(true)
  })

  it('support string variant', () => {
    const wrapper = mount(Divider, {
      props: {
        variant: 'dotted',
      },
      slots: {
        default: `test`,
      },
    })
    expect(wrapper.find('.ant-divider-dotted').exists()).toBe(true)
  })

  it('support vertical size', () => {
    const wrapper1= mount(Divider, {
      props: {
        size: 'small',
      },
      slots: {
        default: `test`,
      },
    })
    expect(wrapper1.find('.ant-divider-sm')).toBeTruthy();

    const wrapper2 = mount(Divider, {
      props: {
        size: 'middle',
      },
      slots: {
        default: `test`,
      },
    })
    expect(wrapper2.find('.ant-divider-md')).toBeTruthy()
  })
})
