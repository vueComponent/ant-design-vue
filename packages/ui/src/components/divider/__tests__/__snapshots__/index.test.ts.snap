// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Divider > should render correctly 1`] = `"<div class="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-center" role="separator"><span class="ant-divider-inner-text">test</span></div>"`;
